didman/api/v1/api_test.go

```diff
@@ -232,6 +232,112 @@ func TestWrapper_AddEndpoint(t *testing.T) {
 	})
 }
 
+func TestWrapper_AddCompoundService(t *testing.T) {
+	id := "did:nuts:1"
+	request := CompoundServiceCreateRequest{
+		Endpoint: map[string]interface{}{
+			"foo": "did:nuts:12345?type=foo",
+			"bar": "did:nuts:54321?type=bar",
+		},
+		Type: "type",
+	}
+
+	t.Run("ok", func(t *testing.T) {
+		ctx := newMockContext(t)
+		var (
+			parsedDID      did.DID
+			parsedEndpoint map[string]ssi.URI
+			parsedType     string
+		)
+		ctx.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
+			p := f.(*CompoundServiceCreateRequest)
+			*p = request
+			return nil
+		})
+		ctx.didman.EXPECT().AddCompoundService(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+			func(subject interface{}, endpointType interface{}, endpoint interface{}) error {
+				parsedDID = subject.(did.DID)
+				parsedEndpoint = endpoint.(map[string]ssi.URI)
+				parsedType = endpointType.(string)
+				return nil
+			})
+		ctx.echo.EXPECT().NoContent(http.StatusNoContent).Return(nil)
+
+		err := ctx.wrapper.AddCompoundService(ctx.echo, id)
+
+		if !assert.NoError(t, err) {
+			return
+		}
+		assert.Equal(t, id, parsedDID.String())
+		assert.Len(t, parsedEndpoint, 2)
+		assert.Equal(t, request.Endpoint["foo"], parsedEndpoint["foo"].String())
+		assert.Equal(t, request.Endpoint["bar"], parsedEndpoint["bar"].String())
+		assert.Equal(t, request.Type, parsedType)
+	})
+
+	t.Run("error - service fails", func(t *testing.T) {
+		ctx := newMockContext(t)
+		ctx.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
+			p := f.(*CompoundServiceCreateRequest)
+			*p = request
+			return nil
+		})
+		ctx.didman.EXPECT().AddCompoundService(gomock.Any(), gomock.Any(), gomock.Any()).Return(errors.New("failed"))
+
+		err := ctx.wrapper.AddCompoundService(ctx.echo, id)
+		test.AssertErrProblemStatusCode(t, http.StatusInternalServerError, err)
+		test.AssertErrProblemDetail(t,  "failed", err)
+		test.AssertErrProblemTitle(t, problemTitleAddCompoundService, err)
+	})
+
+	t.Run("error - incorrect endpoint", func(t *testing.T) {
+		ctx := newMockContext(t)
+		ctx.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
+			p := f.(*CompoundServiceCreateRequest)
+			*p = CompoundServiceCreateRequest{Type: "type", Endpoint: map[string]interface{}{"foo": ":"}}
```

Comment:
what happens if you enter an object or array instead of ":"?

Reply:
Now fixed & tested